main.go

```diff
@@ -22,11 +25,33 @@ type Service struct {
 func (svc *Service) getProtectedResource(c *gin.Context) {
 	authField, authFieldIsPresent := c.Request.Header["Authorization"]
```

Comment:
maybe we can parse this header and extract the macaroon and the preimage in a utils function.

e.g. 
```
macaroon, preimage, err = utils.parseLsatHeader(c.Request.Header["Authorization"])
```
then we don't need to do the TrimSpace and such in the different methods.